robot/impl/local_robot_test.go

```diff
@@ -1931,28 +1931,37 @@ func TestConfigPackageReferenceReplacement(t *testing.T) {
 	test.That(t, r.Close(context.Background()), test.ShouldBeNil)
 }
 
-// removeBuiltinServices removes services with a "builtin" name for testing
-// purposes.
-func removeBuiltinServices(cfg *config.Config) *config.Config {
+// removeDefaultServices removes default services and returns the removed
+// services for testing purposes.
+func removeDefaultServices(cfg *config.Config) []resource.Config {
 	if cfg == nil {
 		return nil
 	}
 
-	var nonBuiltInSvcs []resource.Config
+	// Make a set of registered default services.
+	registeredDefaultSvcs := make(map[resource.Name]bool)
+	for _, name := range resource.DefaultServices() {
+		registeredDefaultSvcs[name] = true
+	}
+
+	var defaultSvcs, nonDefaultSvcs []resource.Config
 	for _, svc := range cfg.Services {
-		if svc.Name != resource.DefaultServiceName {
-			nonBuiltInSvcs = append(nonBuiltInSvcs, svc)
+		if registeredDefaultSvcs[svc.ResourceName()] {
+			defaultSvcs = append(defaultSvcs, svc)
+			continue
 		}
+		nonDefaultSvcs = append(nonDefaultSvcs, svc)
 	}
-	cfg.Services = nonBuiltInSvcs
-	return cfg
+
+	cfg.Services = nonDefaultSvcs
+	return defaultSvcs
 }
 
 func TestConfigMethod(t *testing.T) {
```

Comment:
`TestConfigMethod` has been updated to test that `localRobot.Config()` correctly handles:
- `Cloud`
- Module's `LogLevel`
- Component builtin implicit dependencies
- Service (remote) builtin implicit dependencies
- Non-default builtin resources
- `Packages`
- `Network`
- `Auth`
- `Debug`
- `DisablePartialStart`